File: app/src/main/java/com/easynfc/menu/MenuFragment.java
package com.easynfc.menu;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.easynfc.R;
import com.easynfc.data.model.Menu;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * A simple {@link Fragment} subclass.
 */
public class MenuFragment extends Fragment implements MenuContract.View {


    @BindView(R.id.recycler)
    RecyclerView recycler;
    private MenuAdapter adapter;
    private MenuContract.Presenter presenter;

    public static final String TAG = "MenuFragment";

    public MenuFragment() {

    }

    public static MenuFragment newInstance(){
        return  new MenuFragment();
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_menu, container, false);
        ButterKnife.bind(this,v);
        adapter = new MenuAdapter(getActivity(),Menu.getMenuList());
        GridLayoutManager mLayoutManager = new GridLayoutManager(getActivity(), 1);  //displays number of cards per row
        recycler.setLayoutManager(mLayoutManager);
        recycler.setItemAnimator(new DefaultItemAnimator());
        recycler.setAdapter(adapter);
        return v;
    }





    @Override
    public void onResume() {
        super.onResume();
        presenter.start();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        presenter.stop();
    }


    @Override
    public void setPresenter(MenuContract.Presenter presenter) {
        this.presenter = presenter;
    }

}

File: app/src/main/java/com/easynfc/ui/menu/MenuFragment.kt
package com.easynfc.ui.menu


import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.easynfc.R
import com.easynfc.ui.mytags.MyTagsFragment
import com.easynfc.ui.read.ReadFragment
import com.easynfc.ui.tags.TagsFragment
import com.vipera.onepay.util.FragmentUtils
import kotlinx.android.synthetic.main.fragment_menu.view.*
import org.jetbrains.anko.sdk25.coroutines.onClick


class MenuFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_menu, container, false)

        view.btnWrite.onClick {
            FragmentUtils.show(activity!!.supportFragmentManager!!,TagsFragment.TAG, R.id.container, null)
        }

        view.btnRead.onClick {
            FragmentUtils.show(activity!!.supportFragmentManager!!,ReadFragment.TAG, R.id.container, null)
        }

        view.btnMyTags.onClick {
            FragmentUtils.show(activity!!.supportFragmentManager!!,MyTagsFragment.TAG, R.id.container, null)
        }

        return view
    }

    companion object {

        val TAG = this::class.java.name

        fun newInstance (): MenuFragment {
            return MenuFragment()
        }
    }

}
